src/libraries/System.Private.CoreLib/src/System/SpanHelpers.T.cs

```diff
@@ -2,13 +2,160 @@
+            // where it's used in the arithmetic operations below to ensure no integer overflow is possible.
+
+            if (numElements == 0)
+            {
+                return; // nothing to do
+            }
+
+            if (typeof(T) != typeof(float) && typeof(T) != typeof(double) && !RuntimeHelpers.IsBitwiseEquatable<T>())
+            {
+                goto CannotVectorize; // it might not be valid to SIMD-spray this value into the backing span
+            }
+
+            if (Vector.IsHardwareAccelerated)
+            {
+                if (numElements < (uint)(Vector<byte>.Count / Unsafe.SizeOf<T>()))
+                {
+                    goto CannotVectorize; // not enough data for even a single run through the vectorized loop
+                }
+
+                Vector<byte> vector;
+
+                if (typeof(T) == typeof(float))
+                {
+                    vector = (Vector<byte>)(new Vector<float>((float)(object)value!));
+                }
+                else if (typeof(T) == typeof(double))
+                {
+                    vector = (Vector<byte>)(new Vector<double>((double)(object)value!));
+                }
+                else
+                {
+                    T tmp = value; // Avoid taking address of the "value" argument. It would regress performance of the loops below.
+                    if (Unsafe.SizeOf<T>() == 1)
+                    {
+                        vector = new Vector<byte>(Unsafe.As<T, byte>(ref tmp));
+                    }
+                    else if (Unsafe.SizeOf<T>() == 2)
+                    {
+                        vector = (Vector<byte>)(new Vector<ushort>(Unsafe.As<T, ushort>(ref tmp)));
+                    }
+                    else if (Unsafe.SizeOf<T>() == 4)
+                    {
+                        vector = (Vector<byte>)(new Vector<uint>(Unsafe.As<T, uint>(ref tmp)));
+                    }
+                    else if (Unsafe.SizeOf<T>() == 8)
+                    {
+                        vector = (Vector<byte>)(new Vector<ulong>(Unsafe.As<T, ulong>(ref tmp)));
+                    }
+                    else
+                    {
+                        Debug.Fail("This should never happen. Did we miss special-casing a bitwise equatable type?");
+                        goto CannotVectorize;
+                    }
+                }
+
+                nuint totalByteLength = numElements * (nuint)Unsafe.SizeOf<T>(); // get this calculation ready ahead of time
+                nuint stopLoopAtOffset = totalByteLength & (nuint)(-Vector<byte>.Count * 2);
+                nuint offset = 0;
+
+                // Loop, writing 2 vectors at a time.
+
+                if (numElements >= 2 * (uint)(Vector<byte>.Count / Unsafe.SizeOf<T>()))
+                {
+                    do
+                    {
+                        Unsafe.WriteUnaligned(ref Unsafe.AddByteOffset(ref Unsafe.As<T, byte>(ref refData), offset), vector);
```

Comment:
Want to hoist these `ref Unsafe.As<T, byte>(ref refData)` in a `ref byte byteRef = ref Unsafe.As<T, byte>(ref refData);` and use `ref byteRef` so it's a bit nicer to read?

Codegen will be the same.